### PR TITLE
add support for reading Y4M input files

### LINKED ISSUE #6: Question regarding loading Y4M/YUV videos
Hello Team,

This is a great tool,

I was trying to test, as you may know often the compression standards uses Y4M/YUV files, and it fails to load them as the header is not having information and the ffprobe fails as it cannot find `nb_frames` as it is usually `N/A` for Y4M files. Same happens for `yuv` files. I couldn't figure out how to feed width and height of images via CLI or JSON file. 

Related question,
1. If we feed any video, will it be using `ffmpeg` (or libavfilter in the heart) to decode videos. 
2. Are we always using pytorch impl for any image manipulation tasks (upscaling/downscaling for eg), in the past when we tested `ffmpeg` impl of conversions of YUV->RGB or vice-versa it was not giving accurate colours. So the standard recommendation was to stick to [`HDRTools`](https://gitlab.com/standards/HDRTools) for conversions of YUV to RGB etc.  I did not experiment with pytorch for HDR image manipulation. I am curious to know which one is responsible for conversions here in the case of YUV feeding. 

Sample: https://media.xiph.org/video/aomctc/test_set/hdr2_2k/MeridianTalk_1920x1080_5994_hdr10.y4m

I believe if we support YUV file ingest where we ask users to input {pixel format, size, bitdepth} might be fastest way to resolve this issue.

### PR DESCRIPTION
**Pull request by cosmin**
pass -count_frames to ffprobe for y4m input since it does not contain the number of frames in header

Fixes #6

### CODE REVIEW

**Comment by vibhoothi** [comment]
I totally missed `-count_frames` , it went top of my head^^,

This is smart way.



On a bit unrelated note, do you reckon when we do `rgb48le` conversions (vf_scale yuv -> rgb),ffmpeg would not touch the codevalues and we good to leave to ffmpeg for Y4Ms?

File: pycvvdp/video_source_file.py [hunk 1]
```diff
@@ -70,7 +70,10 @@ class video_reader:
 
     def __init__(self, vidfile, frames=-1, resize_fn=None, resize_height=-1, resize_width=-1, verbose=False):
         try:
-            probe = ffmpeg.probe(vidfile)
+            if vidfile.lower().endswith('.y4m'):
+                probe = ffmpeg.probe(vidfile, count_frames=None)
```
Comment (vibhoothi) [inline pycvvdp/video_source_file.py:74]: We could simply have `count_frames` by default right, so if there are any other ingest supported by ffmpeg and not y4m which do not have frame_count by default, would the software support it? The overhead wouldn't be much and would be cleaner?

Comment (cosmin) [reply pycvvdp/video_source_file.py:74]: I'm not opposed to doing it by default, the overhead of counting frames is a rounding error compared to running ColorVideoVDP. 

I just wanted to make the change as minimally invasive as needed to solve the problem, but happy to adjust.

**Review by mantiuk — APPROVED** [review]
This looks like a reasonable solution, which should not break anything. I need to test whether we can use `count_frames` with other file formats.

**Comment by vibhoothi** [comment]
I did test for prores streams, and it was substantially slower than
defaults. So I would say this makes more sense


Vibhoothi,
Sent from Mobile!
mindfreeze.videolan.me

**Pull request merged**